### PR TITLE
Facepilebutton styles

### PR DESCRIPTION
**Pull request by kkjeer**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Fixes the precedence for FacepileButton styling so that FacepileButtons can be styled by both a `styles` and a `className` prop. Styles are applied to FacepileButton in the following order:

1. Base button styles
2. className prop (styles applied from the FacepileComponent, can be customized using the Facepile styles prop, e.g. <Facepile styles={ { descriptiveOverflowButton: {...} } } />
3. styles prop (styles applied from the FacepileComponent button props, e.g. <Facepile overflowButtonProps={ { styles: { root: {...}, rootHovered: {...}, ... } } } />

The `styles` prop overrides any `className` prop styles, which overrides the base button styles.

Also added a Screener test that includes custom button styles:
screen shot 2018-10-04 at 2 36 37 pm

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by betrue-final-final** [comment]
Hey @kkjeer, what you're doing is showing a custom style, but not the default, correct? Just checking on the screener test.

**Comment by kkjeer** [comment]
> Hey @kkjeer, what you're doing is showing a custom style, but not the default, correct? Just checking on the screener test.

Yes, the new screener test is showing custom styles on the add and overflow buttons in order to test the logic of applying styles to buttons. These are not the default styles.

File: packages/office-ui-fabric-react/src/components/Facepile/FacepileButton.tsx [hunk 1]
```diff
@@ -11,8 +11,18 @@ export class FacepileButton extends BaseComponent<IButtonProps, {}> {
...
+    const { theme, className, styles } = this.props;
 
-    return <BaseButton {...this.props} variantClassName={className} styles={getStyles(theme!, styles)} onRenderDescription={nullRender} />;
+    const customStyles = getStyles(theme!, className, styles);
+
+    return (
+      <BaseButton
+        {...this.props}
+        variantClassName="ms-Button--facepile"
+        className=""
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/Facepile/FacepileButton.tsx]: Why do we need `className=""`?

Comment (kkjeer) [reply packages/office-ui-fabric-react/src/components/Facepile/FacepileButton.tsx]: BaseButtonClassNames styling has the className styles override the styles prop. If the className props is nonempty here, then any className styles (e.g. className="descriptiveOverflowButton") from the parent Facepile component will override any styles from the `styles` prop.

Comment (Vitalius1) [reply packages/office-ui-fabric-react/src/components/Facepile/FacepileButton.tsx]: You mean you try kill the `className` coming from the `{...this.props}` above it? If yes why not filter it out before returning and might as well filter out `styles` as you are overriding it anyway.

**Pull request merged**